Sources/Core/ObjectiveCIR.swift

```diff
@@ -74,6 +74,58 @@ func enumTypeName(propertyName: String, className: String) -> String {
     return "\(className)\(Languages.objectiveC.snakeCaseToCamelCase(propertyName))"
 }
 
+// ObjC integreal types for NS_ENUM declarations in generated code.
+enum EnumerationIntegralType: String {
+    case char
+    case unsignedChar = "unsigned char"
+    case short
+    case unsignedShort = "unsigned short"
+    case int
+    case unsignedInt = "unsigned int"
+    case NSInteger
+    case NSUInteger
+
+    // Return the best fitting and smallest EnumerationIntegralType for the given EnumType.
+    // In ObjC not all NS_ENUM declarations need to be NSInteger in size. Using smaller integral types
+    // can allow the code that plank generates to use less memory in the resulting application.
+    // As an example with an object with 2 enumerations values, each of NSInteger size,
+    // those 2 enumerations will take 16 bytes of the objects instance size in the heap. However,
+    // if the enumeration storage is smaller, in this case 1 byte long for an unsigned char, the
+    // 2 enumerations will only take 8 bytes in the heap. This is because the compiler will best fit
+    // the two unsigned char enumerations to fit into the 8 bytes natural alignment of the platform.
+    // As more enumerations are found in a class, the better this best fitting code will save memory.
+    static func enumerationIntegralTypeFor(_ values: EnumType) -> EnumerationIntegralType {
```

Comment:
I'd consider shortening this name to something like `.forValue`, but that's up to your preferences.

Reply:
I like that.

Reply:
done